hanish/bot.py

```diff
@@ -0,0 +1,409 @@
+            self.darksky.n_api_calls
+        )
+        self.post(msg['channel'], response)
+
+    def handle_unknown_command(self, msg):
+        """
+        Have the chatbot respond when an unknown command is sent.
+        """
+        response = (
+            "I'm happy to chat about the weather, "
+            " unfortunately I don't understand what you're asking."
+        )
+
+        self.post(msg['channel'], response)
+
+    def post(self, channel, response):
+        """
+        Helper function to post a message as the slackbot.
+        """
+        self.slack.api_call(
+            "chat.postMessage", channel=channel, text=response, as_user=True
+        )
+
+    def set_schedule(self):
+        """
+        Sets the schedule to do tasks on demand.
+        """
+        schedule.every().day.at("9:00").do(self.weather_notice)
+
+    def weather(self, zipcode=None):
+        """
+        Quick lookup of the current weather for a given zipcode. If no zipcode
+        is supplied, then will look up the weather for the default zipcode.
+
+        Parameters
+        ----------
+        zipcode: string, default None
+            Zip Code to lookup weather for or None for the default
+
+        Returns
+        -------
+        data: json
+            Prints a nicely idented summary of the weather
+
+        msg: string
+            Returns a string of the number of dark sky api calls made.
+        """
+        # Resolve the latitutde and longitude from the zipcode and query
+        # weather. NOTE: this will utilize cacheing on the API if available.
+        zipcode  = zipcode or self.zipcode
+        lat, lon = self.zipdb.lookup(zipcode)
+        forecast = self.darksky.forecast(lat, lon)
+
+        # Add additional information and return
+        forecast['zipcode'] = zipcode
+        forecast['api_calls'] = self.darksky.n_api_calls
+        return forecast
+
+    def weather_notice(self):
+        """
+        Detects if there is any material change in the weather and alerts the
+        channel. This method is scheduled to run every day at 9:00am
+        """
+        # Keep track if we should alert or not, and get the current weather.
+        alert   = False
+        weather = self.weather()
+
+        if self.yesterday:
+            # Fetch weather for yesterday and today
+            current = weather['hourly']['summary']
+            yesterday = self.yesterday['hourly']['summary']
+
+            # Detect if there has been a change
+            # TODO: Define material change and update accordingly
+            if current != yeseterday:
+                alert = True
+        else:
+            alert = True
+
+        # If alert, post a message.
+        if alert:
+            msg = "Good morning, @channel! A quick update on the weather. "
+            msg += weather_currently(weather)
+            self.post("#general", msg)
+
+        # Save today's weather for tomorrow
+        self.yesterday = weather
```

Comment:
I think I would need more information about the DarkSky API, but no messages would be posted unless the daily forecast is different from "yesterday's" forecast. So it depends on what you mean by problem; if it is scheduled to run more than once per day, right now only the first scheduled run would do anything -- if you wanted multiple alerts per day, then this would be a problem.